### PR TITLE
chore(build): Add WebWorker bundle.

### LINKED ISSUE #3207: Monkey patch window object from within angular for WebWorkers
Currently our WebWorker examples use loader.js to monkey patch the window object. This needs to happen within angular so that user's don't have to do it.

This is slightly complicated because we depend on libraries that expect the window object to already be patched before they're loaded (zone.js specifically. Maybe others)

### PR DESCRIPTION
**Pull request by jteplitz**
Closes #3207

View an example of using the WebWorker bundle [here](https://github.com/jteplitz602/angular_web_worker_example)

### CODE REVIEW

File: docs/typescript-definition-package/index.js [hunk 1]
```diff
@@ -6,107 +6,111 @@ var gitPackage = require('dgeni-packages/git');
...
-  templateFinder.templatePatterns = [
-    '${ doc.template }',
-    '${ doc.id }.${ doc.docType }.template.html',
-    '${ doc.id }.template.html',
-    '${ doc.docType }.template.html',
-    'common.template.html'
-  ];
-});
\ No newline at end of file
+module.exports =
```
Comment (yjbanov) [inline docs/typescript-definition-package/index.js]: could you preserve the formatting (or make it a separate commit)? It's hard to tell where the substantial changes are.

File: modules/examples/src/web_workers/images/loader.js [hunk 2]
```diff
@@ -1,17 +1,4 @@
 $SCRIPTS$
-
```
Comment (yjbanov) [inline modules/examples/src/web_workers/images/loader.js]: Woohoo!

**Comment by yjbanov** [comment]
Travis doesn't like this PR. Also, check with @IgorMinar and @jeffbcross w.r.t. idiomaticity (if that's a word) of bundling and `gulpfile.js` changes.

File: docs/typescript-definition-package/index.js [hunk 3]
```diff
@@ -48,24 +50,38 @@ module.exports = new Package('angular-v2-docs', [jsdocPackage, nunjucksPackage,
...
         id: 'angular2/angular2',
         references: ['../es6-promise/es6-promise.d.ts', '../rx/rx.d.ts'],
-        namespace: 'ng',
         modules: {
-          'angular2/angular2': 'angular2/angular2',
+          'angular2/angular2': {
+            namespace: 'ng',
+            id: 'angular2/angular2'
+          },
+          'angular2/web_worker/worker': {
```
Comment (jeffbcross) [inline docs/typescript-definition-package/index.js]: Why doesn't web_worker get its own typings file?

Comment (jeffbcross) [inline docs/typescript-definition-package/index.js]: @alexeagle was this your idea?

**Comment by jeffbcross** [comment]
LGTM presuming @yjbanov generally approves of internal web worker changes, and presuming @alexeagle endorses the inclusion of web worker typings in angular2.

FWIW I will probably change around the core imports where there will be a `core_no_dom` bundle that has everything but the DOM-related render bits, which would be useful for web_worker
